nix/openssl_1_0_2.nix

```diff
@@ -11,9 +11,25 @@ pkgs.stdenv.mkDerivation rec {
 
   buildInputs = [ pkgs.gnumake pkgs.perl534 ];
 
-  configurePhase = ''
-    ./config -d shared -g3 -fPIC no-libunbound no-gmp no-jpake no-krb5 no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-store no-zlib no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia no-bf no-ripemd no-dsa no-ssl2 no-capieng -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS --prefix=$out
-  '';
+  configurePhase = let
+    default_options =
+      "shared -g3 -fPIC no-libunbound no-gmp no-jpake no-krb5 no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-store no-zlib no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia no-bf no-ripemd no-dsa no-ssl2 no-capieng -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS --prefix=$out";
+  in {
+    x86_64-linux = ''
+      ./config -d ${default_options}
+    '';
+    aarch64-linux = ''
+      ./config ${default_options}
+    '';
```

Comment:
do we need the `-d` for x86?

Can you add a comment about the `-d` and the bug in openssl configure script as comments since thats what most of us will be reading. You can link to this PR for more info

Reply:
We could make it consistent and just drop all the platform filtering, but I was trying not to affect x86 with any of these changes. Adding more words...